Description: Fix ‘givaro_bool_constant’ does not name a type error.
Origin: https://github.com/Macaulay2/M2/commit/ff0f0cb (reversed)
Bug: https://github.com/Macaulay2/M2/issues/2347
Last-Update: 2021-12-18

--- a/M2/Macaulay2/e/aring-gf-givaro.hpp
+++ b/M2/Macaulay2/e/aring-gf-givaro.hpp
@@ -40,7 +40,6 @@
 };
 
 #else
-#define bool_constant givaro_bool_constant
 #include <givaro/gfq.h>
 #include <givaro/givpower.h>
 #include <givaro/givtimer.h>
@@ -50,7 +49,7 @@
 #include <givaro/givintnumtheo.h>
 #include <givaro/givpower.h>
 #include <givaro/givpoly1padic.h>
-#undef bool_constant
+
 #include <type_traits>
 
 namespace M2 {
--- a/M2/Macaulay2/e/aring-zzp-ffpack.hpp
+++ b/M2/Macaulay2/e/aring-zzp-ffpack.hpp
@@ -17,9 +17,8 @@
 
 // this fixes a weird problem in the package "openblas" of Arch Linux, which somehow fails to declare this function:
 extern "C" void openblas_set_num_threads(int num_threads);
-#define bool_constant givaro_bool_constant
+
 #include <fflas-ffpack/ffpack/ffpack.h>
-#undef bool_constant
 #pragma GCC diagnostic pop
 
 namespace M2 {
